src/go/rpk/pkg/cli/cluster/storage/status-mount.go

```diff
@@ -0,0 +1,76 @@
+// Copyright 2024 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+package storage
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/adminapi"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/config"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/out"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+)
+
+func newMountStatus(fs afero.Fs, p *config.Params) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "status-mount [MIGRATION ID]",
+		Short:   "Status of a mount/unmount operation on a topic to the Redpanda Cluster from Tiered Storage",
+		Long:    `Status of a mount/unmount operation on a topic to the Redpanda Cluster from Tiered Storage.`,
+		Aliases: []string{"status-unmount"},
+		Example: `
+Status for a mount/unmount operation
+	rpk cluster storage status-mount 123
+`,
+		Args: cobra.ExactArgs(1),
+		Run: func(cmd *cobra.Command, from []string) {
+			pf, err := p.LoadVirtualProfile(fs)
+			out.MaybeDie(err, "rpk unable to load config: %v", err)
+			f := p.Formatter
+			if h, ok := f.Help(migrationState{}); ok {
+				out.Exit(h)
+			}
```

Comment:
Ideally, we want this at the start of the command since we don't need to load the profile to show the help text.